governance/src/agent.js

```diff
@@ -0,0 +1,301 @@
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      id: voteInfo.id.toString(),
+      address,
+      voter: voteInfo.voter,
+      weight: voteInfo.votingPower.toString(),
+    },
+  });
+}
+
+function proposalCanceledFinding(proposalId, address, config) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Canceled`,
+    description: `Governance proposal ${proposalId} has been canceled`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-PROPOSAL-CANCELED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      id: proposalId,
+      state: 'canceled',
+    },
+  });
+}
+
+function proposalExecutedFinding(proposalId, address, config) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Executed`,
+    description: `Governance proposal ${proposalId} has been executed`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-PROPOSAL-EXECUTED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      id: proposalId,
+      state: 'executed',
+    },
+  });
+}
+
+function proposalQueuedFinding(proposalId, address, config, eta) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Proposal Queued`,
+    description: `Governance proposal ${proposalId} has been queued`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-PROPOSAL-QUEUED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      id: proposalId,
+      state: 'queued',
+    },
+  });
+}
+
+function quorumNumeratorUpdatedFinding(address, config, oldNum, newNum) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Quorum Numerator Updated`,
+    description: `Quorum numerator updated from ${oldNum} to ${newNum}`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-GOVERNANCE-QUORUM-NUMERATOR-UPDATED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      oldNumerator: oldNum,
+      newNumerator: newNum,
+    },
+  });
+}
+
+function timelockChangeFinding(address, config, oldAddress, newAddress) {
+  return Finding.fromObject({
+    name: `${config.protocolName} Governance Timelock Address Change`,
+    description: `Timelock address changed from ${oldAddress} to ${newAddress}`,
+    alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-GOVERNANCE-TIMELOCK-ADDRESS-CHANGED`,
+    type: 'Info',
+    severity: 'Info',
+    protocol: config.protocolName,
+    metadata: {
+      address,
+      oldTimelockAddress: oldAddress,
+      newTimelockAddress: newAddress,
+    },
+  });
+}
+
+function votingDelaySetFinding(address, config, oldDelay, newDelay) {
```

Comment:
this function was never used - you could add a case statement on line 261-284 where you are filtering for log events and create a new case/alert for this